Add 'aws_cloudtrail_service_account' data source

ewbankkit opened the pull request:
New data source `aws_cloudtrail_service_account` to return the account ID and account ARN of the AWS CloudTrail service in the specified region.
See:
* http://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-supported-regions.html

Acceptance tests:
```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudTrailServiceAccount_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSCloudTrailServiceAccount_ -timeout 120m
=== RUN   TestAccAWSCloudTrailServiceAccount_basic
--- PASS: TestAccAWSCloudTrailServiceAccount_basic (29.16s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	30.209s
```

Ninir commented on aws/data_source_aws_cloudtrail_service_account.go:
```diff
@@ -0,0 +1,57 @@
+
+	if accid, ok := cloudTrailServiceAccountPerRegionMap[region]; ok {
+		d.SetId(accid)
+		d.Set("arn", fmt.Sprintf("arn:%s:iam::%s:root", meta.(*AWSClient).partition, accid))
```
Would probably be better to use https://github.com/terraform-providers/terraform-provider-aws/blob/master/vendor/github.com/aws/aws-sdk-go/aws/arn/arn.go#L28, thoughts?

ewbankkit replied:
Agreed.
I added a new `arn.go` file and associated functions in https://github.com/terraform-providers/terraform-provider-aws/pull/1775 which I could use here if that's merged and I rebase.
Also mentioned in https://github.com/terraform-providers/terraform-provider-aws/issues/1783.

Ninir replied:
Agreed, as it has been merged, should be all ok!

Ninir commented on website/docs/d/cloudtrail_service_account.html.markdown:
```diff
@@ -0,0 +1,60 @@
+  Get AWS CloudTrail Service Account ID for storing trail data in S3.
+---
+
+# aws\_cloudtrail\_service\_account
```
These antislashes are not needed anymore when writing markdown

ewbankkit replied:
OK, I'll correct.

Ninir commented on website/docs/d/cloudtrail_service_account.html.markdown:
```diff
@@ -0,0 +1,60 @@
+	"Version": "2008-10-17",
+	"Statement": [
+		{
+        			"Sid": "Put bucket policy needed for trails",
```
Can you fix indentations there? 2 spaces would be really nice :)

ewbankkit replied:
Will do.

ewbankkit commented:
Rebased from `master` and squashed commits.
Re-ran acceptance tests:
```
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSCloudTrailServiceAccount_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -run=TestAccAWSCloudTrailServiceAccount_ -timeout 120m
=== RUN   TestAccAWSCloudTrailServiceAccount_basic
--- PASS: TestAccAWSCloudTrailServiceAccount_basic (19.89s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	20.919s
```
